sql/moz-fx-data-shared-prod/search_derived/search_clients_daily_v8/query.sql

```diff
@@ -20,9 +20,43 @@ CREATE TEMP FUNCTION add_access_point(
   ARRAY(SELECT AS STRUCT CONCAT(key, '.', access_point) AS key, value, FROM UNNEST(entries))
 );
 
-WITH combined_access_point AS (
+-- List of Ad Blocking Addons
+WITH adblocker_addons AS (
+  SELECT
+    addon_id,
+    addon_name
+  FROM
+    `moz-fx-data-shared-prod.telemetry.addon_names`
+  WHERE
+    LOWER(addon_name) LIKE "%block%"
+    AND occurrences > 10000
```

Comment:
I think putting them in a static table would be better. I'm seeing some productivity addons in the list too. @curtismorales - where would be a good place to put the table? I add entries to the table.

Reply:
Identifying which addons block ads in a way that interferes with Mozilla's ability to monetize is not so easy. Here are a few ways we could go about this:

1. Make a static table of addons that (1) occur > 10,000 times (2) contain the word "block" in their name (3) contain the word "track" in their description. (Description listed on [website](https://addons.mozilla.org/en-US/firefox/addon/ublock-origin/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search)).

2. Make a static table of addons that (1) occur > 10,000 times (2) are tagged as "anti-tracker [here](https://addons.mozilla.org/en-US/firefox/tag/anti%20tracker/?utm_content=search&utm_medium=referral&utm_source=addons.mozilla.org).

@dzeber do you have any thoughts?

Reply:
How sensitive do we think this list is? If we're specifically narrowing down to add-ons that interfere with monetization through specific search partners, I'd imagine that's probably sensitive enough to put in a private table? If not, we can put it in static.

You all are the experts here so I defer to you on the criteria we should use. I think if we narrow it down to a certain subset of a hundred or so possibly relevant >10,000 occurrence add-ons, the best thing is probably to just go through manually and pick the ones we want to count.

Reply:
+1 to the idea of a static table - this avoids having metric definitions (and the set of included clients) shifting due to dynamic changes in the add-ons list.

+1 also to the idea of manual curation. We can use one of @rebecca-burwei's rules to create a list of candidates and then review them manually. AFAICT rule 2 gives a list of ~20 candidates (contained on the first AMO result page at the link above). Also, if we have a query that generates the candidate list, we can rerun it periodically and redo manual inspection if there are substantial changes.

I'm not sure what is the range of add-on behaviours to be on lookout for. Rebecca mentioned the "anti-tracker" tag, there is also an ["ad blocker"](https://addons.mozilla.org/en-US/firefox/tag/ad%20blocker/) tag with a lot of overlap. I would lean towards searching the tags rather than the add-on names. Either way, the 10K cutoff is a good limiting factor for keeping the list manageable.

As suggested, I would keep this table private unless there's a good reason not to.

Reply:
This list looks great - it should cover the vast majority of cases. I've added some notes and some suggestions for a couple of other add-ons.

I looked through the metadata for some of these add-ons on AMO, and it looks like not that many actually use the tags feature - sorry, I thought this was more common. However, all add-ons belong to a category, and the add-ons relevant to us are all under Privacy & Security.

After looking through some of the blocking & tracking-related add-ons on AMO I see a few categories emerging:
- dedicated ad blockers: likely affect search monetization
- add-ons from "security suite"/antivirus software that say they do ad blocking: not sure to what extent these will affect monetization, but should probably be on this list.
- add-ons that block or affect cookies (eg. containers): may affect what ads get shown indirectly but don't block ads or attribution. These probably should not be included.
- add-ons that mangle the SERP link URL params: likely affect attribution through tags & should be on the list
- add-ons that say they "block trackers": unless they explicitly block ads, not sure whether this stops SERP ads being shown.

The spreadsheet takes already takes all of this into account.

Reply:
Thanks @dzeber for all of your help! I have documented your advice and code [here](https://github.com/mozilla/search-adhoc-analysis/tree/master/monetization-blocking-addons), so we can refer to it if we need to update the list.

@curtismorales I have updated the SQL to use a private table. Could you help with creating a place for the private table to live? The data that should go in the table is here: https://docs.google.com/spreadsheets/d/1rLDmZoSbXbWHW4jXHACas2ws5FWyfUCedVp3WA_dQrk/edit#gid=1980073953